src/beacons/use-beacons.ts

```diff
@@ -111,10 +111,11 @@ export const useBeacons = () => {
     }
 
     if (granted) {
-      await storage.set(storeKey.beaconsConsent, "true");
       // Initialize beacons SDK after consent is granted
-      await initializeBeaconsSDK();
-      Kettle.grant(BEACONS_CONSENTS);
+      await initializeBeaconsSDK().then(async () => {
+        Kettle.grant(BEACONS_CONSENTS);
+        await storage.set(storeKey.beaconsConsent, "true");
```

Comment:
Make sure is granted only when the initialization went OK and granted also went OK, before it just updated this value first and then the initialization failed, letting the app in a wrong state.